### ISSUE TITLE
Invalid SSL Certificate error in Steam client

### ISSUE THREAD

**Issue by Shished** [flathub/com.valvesoftware.Steam#526]
**Game information**

Steam client, latest version
https://store.steampowered.com/app/753/

**Distribution name and version where applicable**

> NAME="Arch Linux"
> PRETTY_NAME="Arch Linux"
> ID=arch
> BUILD_ID=rolling
> ANSI_COLOR="0;36"
> HOME_URL="https://www.archlinux.org/"
> DOCUMENTATION_URL="https://wiki.archlinux.org/"
> SUPPORT_URL="https://bbs.archlinux.org/"
> BUG_REPORT_URL="https://bugs.archlinux.org/"
> LOGO=archlinux

**Flatpak info**

flatpak --version
`Flatpak 1.6.0`

flatpak --gl-drivers
```
nvidia-440-44
default
host
```

**Problem description**

Steam client throws "Invalid SSL Certificate" error when trying to open any https website in it.
Non-secure websites loads without problems. Everything works fine in native firefox or chromium browsers.

Terminal output shows this errors:

> [0124/171344.841501:ERROR:cert_verify_proc_nss.cc(969)] CERT_PKIXVerifyCert for steamcommunity.com failed err=-8179
> [0124/171344.841896:ERROR:ssl_client_socket_impl.cc(941)] handshake failed; returned -1, SSL error code 1, net_error -202

**Does this issue reproduce with native Steam**

Have not tested

**Comment by nanonyme** [comment]
Does the access work with Chromium in your host?

**Comment by nanonyme** [comment]
The runtime is supposed to be dispatching NSS certificate requests through p11kit to your host.

**Comment by Shished** [comment]
I'm using non-flatpak version of chromium right now and it works fine.
Some ca-certificates packages got updated today.

**Comment by nanonyme** [comment]
Which version of p11kit do you have installed on your host?

**Comment by guihkx** [comment]
Just noticed this error on my Steam client as well. I'm also on Arch Linux, plus I'm on the beta branch of Steam.

>Which version of p11-kit do you have installed on your host?

0.23.19

**Comment by Shished** [comment]
Same version.

**Comment by Erick555** [comment]
So, recent update of p11-kit (0.23.19+)/nss/ca-certificates in Arch break flatpak from reading ssl certificates from host. This may need patching in runtime.

More info:

https://bugzilla.mozilla.org/show_bug.cgi?id=1465613

https://github.com/p11-glue/p11-kit/commit/1def8077a2bc1fc2a6bd3685a9d94a9a51f40e23

https://src.fedoraproject.org/fork/ueno/rpms/ca-certificates/c/6aec97d9bdcbb92ef912a60b86060f4bc6481b1a

https://git.archlinux.org/svntogit/packages.git/commit/trunk?h=packages/nss&id=49ab909853249c416be0918ebe40a8effc938ff1

**Comment by loisgomez** [comment]
Affected by this same issue

**Comment by jekanev** [comment]
same issue with https connection in flatpak apps, qutebrowser etc. Arch distro.

**Comment by nanonyme** [comment]
Yes, this is a generic issue not related to this app.

**Comment by guihkx** [comment]
Are the main developers of Flatpak aware of this issue, or should we report it somewhere else?

**Comment by nanonyme** [comment]
https://gitlab.com/freedesktop-sdk/freedesktop-sdk/issues is a reasonable place to report the issue to.

**Comment by guihkx** [comment]
Thanks.

There's apparently an ongoing tentative of updating p11-kit:

https://gitlab.com/freedesktop-sdk/freedesktop-sdk/merge_requests/2247

**Comment by sdahdah** [comment]
Also being discussed here FYI: https://bbs.archlinux.org/viewtopic.php?id=252390

**Comment by valentindavid** [comment]
Managed to get it to work with: https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/merge_requests/2255

**Comment by valentindavid** [comment]
p11-kit 0.23.20 is going to be released reverting the breaking changes.

**Comment by Pixelnarium** [comment]
Can confirm. Installing testing/p11-kit and multilib-testing/lib32-p11-kit in ArchLinux fixes the problem.

**Comment by Shished** [comment]
p11-kit is updated to 0.23.20 in arch repo and this problem no longer appears.

**Comment by piotr-szegda** [comment]
When we can expect this fix in 20.03? Because I still have this problem in 20.03.

**Comment by thoughton** [comment]
Hi - apologies for commenting on a closed issue, but I am still experiencing this "Invalid SSL" issue in `Ubuntu 20.10` despite (I think) having the upgraded `p11-kit` packages as suggested.

Here's some package info:
```
~$ apt search p11-kit

libp11-kit-dev/groovy-updates,groovy-security,now 0.23.21-2ubuntu0.1 amd64 [installed]
  library for loading and coordinating access to PKCS#11 modules - development

libp11-kit0/groovy-updates,groovy-security,now 0.23.21-2ubuntu0.1 amd64 [installed]
  library for loading and coordinating access to PKCS#11 modules - runtime

p11-kit/groovy-updates,groovy-security,now 0.23.21-2ubuntu0.1 amd64 [installed]
  p11-glue utilities

p11-kit-modules/groovy-updates,groovy-security,now 0.23.21-2ubuntu0.1 amd64 [installed]
  p11-glue proxy and trust modules
```
as can be seen, all the packages are installed at version `0.23.21`.

Are there any special instructions for fixing this issue on Ubuntu?

Thanks!
